cmds/webboot/network.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/u-root/NiChrome/pkg/wifi"
+	"github.com/u-root/webboot/pkg/menu"
+	"github.com/vishvananda/netlink"
+)
+
+func connected() bool {
+	client := http.Client{
+		Timeout: 10 * time.Second,
+	}
+
+	if _, err := client.Get("http://google.com"); err != nil {
+		return false
+	}
+	return true
+}
+
+func interfaceEntries() ([]menu.Entry, error) {
+	interfaces, err := netlink.LinkList()
+	if err != nil {
+		return nil, err
+	}
+
+	var ifEntries []menu.Entry
+	for _, iface := range interfaces {
+		ifEntries = append(ifEntries, &Interface{label: iface.Attrs().Name})
+	}
+	return ifEntries, nil
+}
+
+func interfaceIsWireless(ifname string) bool {
+	devPath := fmt.Sprintf("/sys/class/net/%s/wireless", ifname)
+	if _, err := os.Stat(devPath); err != nil {
+		return false
+	}
+	return true
+}
+
+func setupNetwork(uiEvents <-chan ui.Event) (bool, error) {
+	iface, err := selectNetworkInterface(uiEvents)
+	if err != nil {
+		return false, err
+	}
+
+	err = selectWirelessNetwork(uiEvents, iface)
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func selectNetworkInterface(uiEvents <-chan ui.Event) (string, error) {
+	ifEntries, err := interfaceEntries()
+	if err != nil {
+		return "", err
+	}
+
+	for {
+		iface, err := menu.DisplayMenu("Network Interfaces", "Choose an option", ifEntries, uiEvents)
+		if err != nil {
+			return "", err
+		}
+
+		if !interfaceIsWireless(iface.Label()) {
+			menu.DisplayResult([]string{"Only wireless network interfaces are supported."}, uiEvents)
+		} else {
+			return iface.Label(), nil
+		}
+	}
+}
+
+func selectWirelessNetwork(uiEvents <-chan ui.Event, iface string) error {
```

Comment:
I am not sure, but did you provide a way for user to input the password of network? it that only support non password wifi now?